pkg/scheduler/framework/plugins/helper/spread.go

```diff
@@ -36,36 +50,43 @@ func DefaultSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.R
 			labelSet = labels.Merge(labelSet, service.Spec.Selector)
 		}
 	}
+	selector := labelSet.AsSelector()
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
-		}
+	owner := metav1.GetControllerOfNoCopy(pod)
+	if owner == nil {
+		return selector
 	}
 
-	selector := labels.NewSelector()
-	if len(labelSet) != 0 {
-		selector = labelSet.AsSelector()
+	gv, err := schema.ParseGroupVersion(owner.APIVersion)
+	if err != nil {
+		return selector
 	}
 
-	if rss, err := rsl.GetPodReplicaSets(pod); err == nil {
-		for _, rs := range rss {
+	gvk := gv.WithKind(owner.Kind)
+	switch gvk.String() {
```

Comment:
why not just using the plain gvk object for comparison purpose:

```
var xyzKind = ...SchemeGroupVersion.WithKind("...")

switch gvk {
```

Reply:
done.